src/domain/bitcoin/onchain/incoming-tx-handler.ts

```diff
@@ -80,3 +82,78 @@ export const IncomingOnChainTxHandler = (
     balanceByWallet,
   }
 }
+
+export const NewIncomingOnChainTxHandler = <S extends WalletCurrency>(
+  txns: WalletOnChainSettledTransaction[],
+): NewIncomingOnChainTxHandler<S> | ValidationError => {
+  const walletCurrency = txns[0].settlementCurrency
+  const settlementCurrencies = new Set(txns.map((tx) => tx.settlementCurrency))
+  if (settlementCurrencies.size !== 1) {
+    return new MultipleCurrenciesForSingleCurrencyOperationError()
```

Comment:
this does not match logic in `briaGetPendingOnChainBalanceForWallets`

Reply:
True, I added some changes to codify that the lnd handler should not need to handle non-BTC-wallet addresses/transactions because we haven't turned on stablesats receives as yet https://github.com/GaloyMoney/galoy/pull/2662/commits/516c2dbfd6a04a694f56b1bf6b435132d9d4ded4

We can decide if we need to do some more work to listen to lnd and add their pendings to pending-repo and remove when we settled those transactions (maybe ACE topic).

Reply:
`IncomingTxHandler` was removed https://github.com/GaloyMoney/galoy/pull/2662/commits/4a7451d19a8c01acf01f36e4410d31cd46bb057a